### PR TITLE
Add environment to diagnose output

### PR DESCRIPTION
**Pull request by tombruijn**
I found that it could be useful to see the environment a person is running diagnose under.

Things like checking the API key usually fails in `development`, because people have disabled AppSignal there. Knowing what env they run the command helps us identify if they should instead run the command with `APPSIGNAL_APP_ENV=production appsignal diagnose` in order to check the API key, for example.

I also updated the spec to avoid the use of an instance variable and not call `#string` for every check.

### CODE REVIEW

File: spec/lib/appsignal/cli/diagnose_spec.rb [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
       out_stream.string.should include('Gem version')
       out_stream.string.should include('Agent version')
+      out_stream.string.should include('Environment')
```
Comment (jeffkreeftmeijer) [inline spec/lib/appsignal/cli/diagnose_spec.rb]: Nitpicking: could you add the actual environment to the assertion here to make sure it's printed correctly, or is there any reason `Appsignal.config.env` is unstable in the tests?

**Comment by jeffkreeftmeijer** [comment]
Test-tasctic! :+1:

File: spec/lib/appsignal/cli/diagnose_spec.rb (hunk 1 above)
Comment (tombruijn) [inline spec/lib/appsignal/cli/diagnose_spec.rb]: That is something I have another branch for. I wanted to commit all tests in one go, but that was a bit more tricky than I thought. There's no test for the mkmf.log output for example now.

**Comment by matsimitsu** [comment]
👍

**Pull request merged**